astroid/brain/brain_pathlib.py

```diff
@@ -18,6 +19,9 @@
 
 
 def _looks_like_parents_subscript(node: nodes.Subscript) -> bool:
+    """Infer subscripted names on Python 3.10+, which supports slicing PurePath.parents"""
+    if not PY310_PLUS:
+        return False
```

Comment:
Better to wrap the `register_transform` call below in a version check. That way we don't even call the predicate function for `< 3.10`.

https://github.com/PyCQA/astroid/blob/91533b7b8a5f7951c2b590add0a73ad3bc113a07/astroid/brain/brain_typing.py#L416-L419